### PR TITLE
feat(cvaa): add TEXT_STYLE_CHANGED event

### PR DESCRIPTION
**Pull request by yairans**
### Description of the Changes
Fire an event on text style changed, mainly for local storage.

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [x] new unit / functional tests have been added (whenever applicable)
- [x] test are passing in local environment 
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/player.js [hunk 1]
```diff
@@ -794,6 +794,7 @@ export default class Player extends FakeEventTarget {
         sheet.insertRule(`#${this._playerId} .${SUBTITLES_CLASS_NAME} > div > div > div { ${style.toCSS()} }`, 0);
       }
       this._textStyle = style;
+      this.dispatchEvent(new FakeEvent(CustomEvents.TEXT_STYLE_CHANGED, {textStyle: style}));
```
Comment (OrenMe) [inline src/player.js]: Don’t emit object by ref, in this case don’t emit the event with payload at all, make the listener of this explicitly need to call player.textStyle to get it and remove this payload as this is not data but an actual class instance object

File: src/player.js [hunk 2]
```diff
@@ -1031,6 +1032,9 @@ export default class Player extends FakeEventTarget {
       if (typeof this._config.playback.playsinline === 'boolean') {
         this.playsinline = this._config.playback.playsinline;
       }
+      if (typeof this._config.playback.textStyle === "object") {
+        this.textStyle = Utils.Object.mergeDeep(new TextStyle(), this._config.playback.textStyle);
```
Comment (OrenMe) [inline src/player.js]: Player should know only TextStyle, if someone serializes class data outside of player he should recreate the TextStyle there and set it as player.textStyle = textStyle

File: src/player.js [hunk 3]
```diff
@@ -794,6 +794,7 @@ export default class Player extends FakeEventTarget {
         sheet.insertRule(`#${this._playerId} .${SUBTITLES_CLASS_NAME} > div > div > div { ${style.toCSS()} }`, 0);
       }
       this._textStyle = style;
+      this.dispatchEvent(new FakeEvent(CustomEvents.TEXT_STYLE_CHANGED, {textStyle: this._textStyle.clone()}));
```
Comment (dan-ziv) [inline src/player.js]: Change the payload to:
` {textStyle: this.textStyle}`
it's already cloned the object,
so why access the private and clone it by yourself?

File: src/player.js (hunk 1 above)
Comment (yairans) [reply src/player.js]: done

File: src/player.js (hunk 2 above)
Comment (yairans) [reply src/player.js]: removed.

File: src/player.js (hunk 3 above)
Comment (yairans) [reply src/player.js]: irrelevant. i've removed the payload. taken from `textStyle` getter.

**Comment by dan-ziv** [comment]
@yairans 
approved just fix tests (you need to add the new event to the list in events.spec)

**Comment by OrenMe** [comment]
@yairans test is failing, please fix

**Comment by OrenMe** [comment]
@dan-ziv it's not clear to me what the test, which is now failing, does?! it just checks that the test array equals the actual player array. I think it holds no value and we can just remove it?

**Pull request merged**